cli/command/container/create_test.go

```diff
@@ -121,6 +122,105 @@ func TestCreateContainerPullsImageIfMissing(t *testing.T) {
 	assert.Check(t, is.Contains(stderr, "Unable to find image 'does-not-exist-locally:latest' locally"))
 }
 
+func TestCreateContainerNeverPullsImage(t *testing.T) {
+	imageName := "does-not-exist-locally"
+	responseCounter := 0
+	pullCounter := 0
+
+	client := &fakeClient{
```

Comment:
I'm still not sure about this test.
Let's keep the handler functions here simple and do what's expected (e.g. image create -> create fake image response) and have them increment the counters we are interested in.
Then when we actually call `createContainer` we check the pull counter to make sure it stays at zero (or whatever we expect based on the input).

I also think we can pull all this into one test like:

```go
func TestCreateContainerImagePullPolicy(t *testing.T) {
   // setup client stuff

  type testCase struct {
    Policy string
    ExpectedPulls int
    ExpectedErr err
  }
  cases := []testCase {...}
  for _, c := range cases {
    t.Run(c.Policy, func(t *testing.T) {
      // createContainer
      // check actual vs expected
    }
  }
}

Reply:
Made a single table test for all the logic.